### PR TITLE
NEXUS-5372: Alternative fix

### PR DESCRIPTION
**Pull request by cstamas**
Introduced new format, to replace the old broken
format, while keeping backward compat. In cases when
records are no longer than 255 bytes at least.

As in real life, records are longer than 255, this does not "salvage" the historical data (that is already written out corrupted), but at least does not break the build. Also, it uses proper Protobuf API to write out delimited records, is not rolling it's own to do so.

### CODE REVIEW

**Comment by ifedorenko** [comment]
-1

Lets remove V1 and V2 formats. V2 was broken and there is no point in providing support for it. V1 was before V2, so any historic timelines customers might have are broken.

Please provide a unit test for records longer than 255 bytes.

**Comment by cstamas** [comment]
Applied Igor's comments.

File: nexus/plugins/nexus-timeline-plugin/src/main/java/org/sonatype/timeline/internal/DefaultTimelinePersistor.java [hunk 1]
```diff
@@ -106,14 +100,12 @@ protected synchronized void persist( final TimelineRecord... records )
         OutputStream out = null;
         try
         {
+            out = new FileOutputStream( getDataFile(), true );
```
Comment (ifedorenko) [inline nexus/plugins/nexus-timeline-plugin/src/main/java/org/sonatype/timeline/internal/DefaultTimelinePersistor.java]: I'd use BufferedOutputStream here.

File: nexus/plugins/nexus-timeline-plugin/src/main/java/org/sonatype/timeline/internal/DefaultTimelinePersistor.java [hunk 2]
```diff
@@ -262,12 +254,12 @@ else if ( filePtr >= result.size() )
         try
         {
             in = new FileInputStream( file );
```
Comment (ifedorenko) [inline nexus/plugins/nexus-timeline-plugin/src/main/java/org/sonatype/timeline/internal/DefaultTimelinePersistor.java]: I'd use BufferedInputStream here

**Comment by ifedorenko** [comment]
+1 with the comments above

**Comment by nabcos** [comment]
+1

**Pull request merged**